### PR TITLE
Ensure `ServerlessError` instances have `code` set

### PR DESCRIPTION
**Pull request by pgrzesik**
Whenever it was fairly strainghforward, I've added tests for added codes, however, in a lot of places that would require modifying old-style test which means extensive test suite refactoring. I decided to skip that in such cases due to how time consuming that would be. If you feel otherwise and think we should first continue with refactoring of tests, please let me know

Addresses: #9346

### CODE REVIEW

File: lib/classes/ConfigSchemaHandler/index.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ class ConfigSchemaHandler {
             ? `${ERROR_PREFIX}: \n     ${messages.join('\n     ')}`
             : `${ERROR_PREFIX} ${messages[0]}`;
 
-        throw new ServerlessError(errorMessage);
+        throw new ServerlessError(errorMessage, 'SCHEMA_VALIDATION');
```
Comment (medikoo) [inline lib/classes/ConfigSchemaHandler/index.js:125]: I'd rather put as `INVALID_NON_SCHEMA_COMPLIANT_CONFIGURATION`

File: lib/classes/Variables.js [hunk 2]
```diff
@@ -451,7 +451,7 @@ class Variables {
         ` a string for variable ${matchedString}.`,
         ' Please make sure the value of the property is a string.',
       ].join('');
-      throw new ServerlessError(errorMessage);
+      throw new ServerlessError(errorMessage, 'INVALID_NON_STRING_PROPERTY');
```
Comment (medikoo) [inline lib/classes/Variables.js:454]: I think it's more `INVALID_NON_STRING_VARIABLE_RESOLUTION_RESULT`

File: lib/configuration/variables/sources/env.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ module.exports = {
     address = ensureString(address, {
       Error: ServerlessError,
       errorMessage: 'Non-string address argument in variable "env" source: %v',
+      errorCode: 'INVALID_ENV_SOURCE_ADDRESS_ARGUMENT',
```
Comment (medikoo) [inline lib/configuration/variables/sources/env.js:17]: I think we can strip `_ARGUMENT` postfix (and same in other cases, technically _address_ and _params_ are an alias for _argument_ term here, so it kind of duplicates information)

File: lib/plugins/aws/lib/monitorStack.js [hunk 4]
```diff
@@ -91,7 +91,7 @@ module.exports = {
                 let errorMessage = 'An error occurred: ';
                 errorMessage += `${stackLatestError.LogicalResourceId} - `;
                 errorMessage += `${stackLatestError.ResourceStatusReason}.`;
-                throw new ServerlessError(errorMessage);
+                throw new ServerlessError(errorMessage, 'STACK_MONITORING_FAILURE');
```
Comment (medikoo) [inline lib/plugins/aws/lib/monitorStack.js:94]: It's rather `STACK_OPERATION_FAILURE` (not really an issue with monitoring)

File: lib/plugins/config.js [hunk 5]
```diff
@@ -123,7 +124,8 @@ class Config {
 
       if (!validShells.has(shell)) {
         throw new ServerlessError(
-          `Shell "${shell}" is not supported. Supported shells: ${Array.from(validShells)}.`
+          `Shell "${shell}" is not supported. Supported shells: ${Array.from(validShells)}.`,
+          'INVALID_SHELL'
```
Comment (medikoo) [inline lib/plugins/config.js:128]: I'd put `TABCOMPLETION_INVALID_SHELL_ARGUMENT`

File: lib/plugins/print.js [hunk 6]
```diff
@@ -100,7 +100,7 @@ class Print {
         conf = conf[step];
 
         if (!conf) {
-          throw new ServerlessError(`Path "${this.options.path}" not found`);
+          throw new ServerlessError(`Path "${this.options.path}" not found`, 'PATH_NOT_FOUND');
```
Comment (medikoo) [inline lib/plugins/print.js:103]: Maybe `INVALID_PATH_ARGUMENT`

File: lib/plugins/standalone.js [hunk 7]
```diff
@@ -57,7 +57,8 @@ module.exports = class Standalone {
           'Note: Service is safe to upgrade if no deprecations are logged during its deployment.',
           `      Check https://github.com/serverless/serverless/releases/tag/v${latestMajor}.0.0 ` +
             'for list of all breaking changes',
-        ].join('\n')
+        ].join('\n'),
+        'CANNOT_UPGRADE_TO_MAJOR_WITHOUT_BREAKING_CHANGES'
```
Comment (medikoo) [inline lib/plugins/standalone.js:61]: I'd put it shorter as `CANNOT_UPGRADE_MAJOR`

**Review by medikoo — CHANGES_REQUESTED** [review]
@pgrzesik looks great 👍 

I've proposed just few improvements, so more-less we stick to one convention.

> I decided to skip that in such cases due to how time consuming that would be

I didn't expect any updates to tests here, they can eventually be done when revisiting given functionalities, and when legacy tests are concerned I would definitely not touch them.

**Comment by pgrzesik** [comment]
Thanks @medikoo for great suggestions, I believe I've updated all of mentioned codes.

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**